### PR TITLE
[bdd-engine] Align fail-fast properties

### LINKED ISSUE #1369: Align naming of fail-fast properties
**Is your feature request related to a problem? Please describe.**
For now we have 3 properties to fail-fast execution stages:
 - `soft-assert.fail-fast=true` - stops the scenario
 - `bdd.configuration.reset-state-before-scenario=false` - stops the story
 - `bdd.ignore-failure-in-batches=false` - stops the batch

**Describe the solution you'd like**
The naming should be revised and unified

### PR DESCRIPTION
**Pull request by ikalinin1**
Closes #1369

### CODE REVIEW

File: docs/modules/ROOT/pages/tests-configuration.adoc [hunk 1]
```diff
@@ -50,13 +50,25 @@ NOTE: The properties marked with *bold* are mandatory.
 |suite meta-filter
 |The meta-filter used to filter the batch stories and scenarios
 
-|`bdd.batch-<batch-number>.ignore-failure`
-|`true`
-|If `false` and any failure is occurred during the batch execution, the subsequent batches are not executed.
+|`bdd.batch-<batch-number>.fail-fasle`
```
Comment (valfirst) [inline docs/modules/ROOT/pages/tests-configuration.adoc:53]: `fasle` -> `fast`

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/batch/BatchExecutionConfiguration.java [hunk 2]
```diff
@@ -28,6 +28,7 @@
     private List<String> metaFilters;
     private Duration storyExecutionTimeout;
     private Boolean ignoreFailure;
```
Comment (valfirst) [inline vividus-bdd-engine/src/main/java/org/vividus/bdd/batch/BatchExecutionConfiguration.java:30]: is it used?

File: vividus/src/main/java/org/vividus/configuration/DeprecatedPropertiesHandler.java [hunk 3]
```diff
@@ -38,11 +44,26 @@
     private final Properties deprecatedProperties;
     private final String placeholderWarnMsg;
     private final Pattern placeholderPattern;
+    private final List<String> keysToFlipBooleanValue;
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/configuration/DeprecatedPropertiesHandler.java:47]: `Flip` -> `Invert`

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus/src/main/java/org/vividus/bdd/model/Statistic.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    {
+        return total.get();
+    }
+
+    public long getPassed()
+    {
+        return passed.get();
+    }
+
+    public void increasePassed()
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/bdd/model/Statistic.java:42]: `increaseXXX` -> `incrementXXX`

File: vividus/src/main/java/org/vividus/configuration/DeprecatedPropertiesHandler.java [hunk 5]
```diff
@@ -51,25 +78,45 @@ public DeprecatedPropertiesHandler(Properties deprecatedProperties, String place
...
+            if (isDepreacted(placeholderKey))
             {
-                warnDeprecated(placeholderWarnMsg, key, placeholderKey, deprecatedProperties.get(placeholderKey));
+                warnDeprecated(placeholderWarnMsg, key, placeholderKey,
+                        getDeprecatedProperty(placeholderKey).getNewKey(placeholderKey));
             }
         }
     }
 
+    private boolean isDepreacted(String key)
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/configuration/DeprecatedPropertiesHandler.java:98]: `isDeprecated `

File: vividus/src/main/java/org/vividus/configuration/DeprecatedPropertiesHandler.java [hunk 6]
```diff
@@ -35,14 +48,28 @@
     private static final String PLACEHOLDER_WARN_MSG_FORMAT =
             "Property '{}' uses deprecated placeholder '%1$s{}%2$s'. Use '%1$s{}%2$s' instead";
 
-    private final Properties deprecatedProperties;
     private final String placeholderWarnMsg;
     private final Pattern placeholderPattern;
+    private final List<DeprecatedProperty> dynamicDeprecatedProperties = new ArrayList<>();
+    private final Map<String, DeprecatedProperty> deprecatedProeprties = new HashMap<>();
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/configuration/DeprecatedPropertiesHandler.java:54]: `deprecatedProperties `

File: vividus/src/main/java/org/vividus/configuration/DeprecatedPropertiesHandler.java [hunk 7]
```diff
@@ -83,20 +130,108 @@ public void replaceDeprecated(Properties checkProperties, Properties replaceInPr
...
+    }
+
     protected void warnDeprecated(String format, Object... arguments)
     {
         LOGGER.warn(format, arguments);
     }
+
+    private final class DynamicDeprecatedProperty extends DeprecatedProperty
+    {
+        private final Splitter dynamicValue = Splitter.on(Pattern.compile("\\(.+?\\)")).omitEmptyStrings();
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/configuration/DeprecatedPropertiesHandler.java:164]: move `Pattern.compile("\\(.+?\\)")` to a new constant

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus/build.gradle [hunk 8]
```diff
@@ -70,6 +70,7 @@ dependencies {
     implementation(group: 'org.hamcrest', name: 'hamcrest', version: versions.hamcrest)
     implementation(group: 'com.google.guava', name: 'guava', version: versions.guava)
     implementation(group: 'io.github.classgraph', name: 'classgraph', version: '4.8.102')
+    implementation(group: 'io.qameta.allure', name: 'allure-model', version: versions.allure)
```
[the remark this answers is not in the document]
Comment (ikalinin1) [reply vividus/build.gradle:73]: Without it, there is a compilation error. I'll investigate

File: vividus/src/main/java/org/vividus/bdd/model/Node.java [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+import java.util.Optional;
+
+import org.vividus.bdd.Status;
+
+public class Node
+{
+    private final NodeType type;
+    private final Deque<Node> children = new LinkedList<>();
+    private Status status;
+    private Optional<Boolean> expectedToHaveChildrens = Optional.empty();
```
[the remark this answers is not in the document]
Comment (ikalinin1) [reply vividus/src/main/java/org/vividus/bdd/model/Node.java:30]: I need third state => not set

[the remark this answers is not in the document]
Comment (valfirst) [reply vividus/src/main/java/org/vividus/bdd/model/Node.java:30]: could you please point where "not set" state is used?

[the remark this answers is not in the document]
Comment (ikalinin1) [reply vividus/src/main/java/org/vividus/bdd/model/Node.java:30]: It was inferred to orElse(Boolean.FALSE) :)

**Pull request closed without merging**